### PR TITLE
chore(demo): Give PostHog team members staff privileges in demo env

### PR DESCRIPTION
**Pull request by Twixes**
## Problem

In the demo environment there's no passwords, but to manage the instance, there needs to be someone with `is_staff=True`. Hence there needs to be _some_ way to log in securely.

## Changes

This leverages login with Google for automatic provisioning of accounts with `is_staff=True`. It's also not possible to log into password-less `is_staff=True` accounts otherwise, so only @posthog.com team members have elevated privileges.

## How did you test this code?

Writing tests…

### CODE REVIEW

File: posthog/api/test/test_signup.py [hunk 1]
```diff
@@ -690,6 +691,35 @@ def test_social_signup(self, *args):
         assert user.is_active is True
         assert user.is_staff is True
 
+    def test_social_login_give_staff_privileges(self, *args):
+        # This creates a user with is_staff=False
+        User.objects.bootstrap("Tech R Us", "charlie@tech-r-us.com", None, first_name="Charlie")
```
Comment (pauldambra) [inline posthog/api/test/test_signup.py:678]: we could replace the comment with `assert user.is_staff is False` after the user bootstrap

File: frontend/src/scenes/organization/ConfirmOrganization/ConfirmOrganization.tsx [hunk 2]
```diff
@@ -21,7 +21,6 @@ export const scene: SceneExport = {
 export function ConfirmOrganization(): JSX.Element {
     const { isConfirmOrganizationSubmitting, email, showNewOrgWarning } = useValues(confirmOrganizationLogic)
     const { setShowNewOrgWarning } = useActions(confirmOrganizationLogic)
-    console.log(showNewOrgWarning)
```
Comment (pauldambra) [inline frontend/src/scenes/organization/ConfirmOrganization/ConfirmOrganization.tsx:24]: 🚀

**Review by pauldambra — APPROVED** [review]
one nit but otherwise looks correct

**Review by pauldambra — APPROVED** [review]

**Comment by Twixes** [comment]
Wow, it was surprisingly tricky to make the `settings`-dependent signup serializer validators testable – mostly because making them `settings`-dependent at the class definition level doesn't work when the class doesn't get re-loaded by the interpreter in the face of `settings` changing. Instead, the validators need to be determined at class runtime.
Also had to learn how 400s with `attr` are made LOL: `ValidationError({ "email": "Error" })`

**Pull request merged**